### PR TITLE
The Find Antibiotics mission should also accept broad-spectrum antibiotics

### PR DESCRIPTION
**Pull request by Zlorthishen**
#### Summary
Summary(Bugfix) "The Find Antibiotics mission should also accept broad-spectrum antibiotics"

#### Purpose of change
The Find Antibiotics mission only accepts accepts vanilla antibiotics rather than all the things that might cure the infection.

#### Describe the solution
changes the mission to accept Strong Antibiotics, Panaceia and regular antibiotics, implements [dda 47011](https://github.com/CleverRaven/Cataclysm-DDA/pull/47011) by [langp11](https://github.com/langp11)

#### Testing
Spawned an NPC with debug command, remove any existing mission. add "Find Antibiotics". Verify that items fulfill mission requirements.

### CODE REVIEW

**Comment by olanti-p** [comment]
Does the antibiotics item get removed from inventory on mission completion?

**Comment by Zlorthishen** [comment]
> Does the antibiotics item get removed from inventory on mission completion?

it does not. what would be a good way of removing them?

**Comment by Coolthulhu** [comment]
Certainly needs that removal bit.

**Comment by olanti-p** [comment]
What if you were instead to set the goal as `MGOAL_FIND_ITEM_GROUP`, define a custom item group that consists of the 3 items, and reference it in mission definition with `"item_group": "the_3_options"`?

**Comment by Zlorthishen** [comment]
> What if you were instead to set the goal as `MGOAL_FIND_ITEM_GROUP`, define a custom item group that consists of the 3 items, and reference it in mission definition with `"item_group": "the_3_options"`?
cataclysm-tiles_2023-04-02_18-40-04

File: data/json/npcs/missiondef.json [hunk 1]
```diff
@@ -62,11 +62,11 @@
     "id": "MISSION_GET_ANTIBIOTICS",
     "type": "mission_definition",
     "name": { "str": "Find Antibiotics" },
-    "goal": "MGOAL_FIND_ITEM",
+    "goal": "MGOAL_FIND_ITEM_GROUP",
     "difficulty": 2,
     "value": 150000,
     "urgent": true,
-    "item": "antibiotics",
+    "item": "drugs_antibiotics",
```
Comment (chaosvolt) [inline data/json/npcs/missiondef.json:69]:
```suggestion
    "item_group": "drugs_antibiotics",
```
Ported JSON changes over and tested, it doesn't seem to recognize any of those items as valid. :/

Comment (Zlorthishen) [reply data/json/npcs/missiondef.json]: we may want to double-test if `MGOAL_FIND_ITEM_GROUP` actually works

**Review by olanti-p — APPROVED** [review]
Can confirm it works together with #2546

**Pull request merged**